### ISSUE TITLE
Events ending at midnight shouldn't be included in the next day

### ISSUE THREAD

**Issue by Pikrass** [pimutils/khal#808]
Let's take an event starting at Monday 20:00 and ending at exactly midnight. Currently, that event would show on the calendar both for Monday and Tuesday.

Some other calendar apps do what I think is the right thing: show the event only for Monday. In fact, I often set the end time to midnight for events I know will span for a few more hours, just so they don't clutter my agenda for the next day.

I can work on this as soon as I have time (which may not happen soon). If you've got some pointers on how to do that... :)

**Comment by Pikrass** [comment]
Oops, this is apparently already fixed, I hadn't upgraded since then.
Sorry!

**Comment by Pikrass** [comment]
Aaaand I noticed this doesn't in fact always work. Seems like it's okay for one-off events but not for repeated ones. I'll try to investigate more when I have time.

**Comment by geier** [comment]
Can you share an event that triggers this bug?

**Comment by Pikrass** [comment]
Okay, I found the culprit. For some reason, for events with an RRULE, my app on Android set a DTSTART with non-zero seconds (probably based on the current time); while for ponctual events the seconds were set to zero.

So that's not really a bug, though that's a bit annoying/surprising when timeformat is set not to include seconds.

**Comment by WhyNotHugo** [comment]
What's the duration of the event? Can you share a full example?

If the issue is that the seconds are set to non-zero, then if might be just that the event actually _does_ end on the next day.

Say, an event at 23:00:01, that lasts 60minutes, actually does end at 00:00:01, and I think working around that kind of scenario is a bit of a hack -- that needs to be fixed on the android app that creates the events.

Is that the case though?

**Comment by Pikrass** [comment]
Yes the duration made the event go past midnight. So khal's behavior is correct, but hard to understand given the
seconds aren't displayed by default, even on the edit screen.

I agree the app on Android should be fixed. That's the default one on Huawei, I don't think it's open source.

The question is, should we make this easier to "troubleshoot"?
At least, we could display the seconds on the edit screen (maybe only when they're non-zero?), so the user can identify
the problem and fix it within khal.
